src/Sylius/Bundle/CoreBundle/Controller/ProductController.php

```diff
@@ -114,6 +115,58 @@ public function indexByTaxonIdAction(Request $request, $id)
         return $this->renderResults($taxon, $paginator, 'productIndex.html', $request->get('page', 1));
     }
 
+    /**
+     * List products categorized under given archetype (fetch by its ID).
+     *
+     * @param Request $request
+     * @param integer $id
+     *
+     * @return Response
+     *
+     * @throws NotFoundHttpException
+     */
+    public function indexByArchetypeIdAction(Request $request, $id)
```

Comment:
`indexByArchetypeAction`? Also It's better (API-friendly) to get `id` (or `archetype`) from $request:
```php
$archetypeId = $request->get('archetype');
```